code/game/machinery/shieldgen.dm

```diff
@@ -553,3 +553,656 @@
+		try_connect(user)
+		return
+
+	if(default_change_direction_wrench(user, I))
+		if(shield_generator)
+			shield_generator.connected_modules -= (src)
+			shield_generator.calculate_boost()
+			shield_generator = null
+			update_icon_state()
+		if(connected_node)
+			connected_node.connected_through_us -= (src)
+			connected_node = null
+		connected_turf = get_step(loc, dir)
+		return TRUE
+
+	if(default_deconstruction_crowbar(I))
+		return TRUE
+	return..()
+
+/obj/machinery/modular_shield/module/setDir(new_dir)
+	. = ..()
+	connected_turf = get_step(loc, dir)
+
+/obj/machinery/modular_shield/module/proc/try_connect(user)
+
+	if(shield_generator || connected_node)
+		balloon_alert(user, "already connected to something")
+		return
+
+	shield_generator = (locate(/obj/machinery/modular_shield_generator) in connected_turf)
+
+	if(shield_generator)
+
+		shield_generator.connected_modules |= (src)
+		balloon_alert(user, "connected to generator")
+		update_icon_state()
+		if(istype(src, /obj/machinery/modular_shield/module/node))
+			var/obj/machinery/modular_shield/module/node/connected_node = src
+			connected_node.connect_connected_through_us()
+		shield_generator.calculate_boost()
+		return
+
+	connected_node	= (locate(/obj/machinery/modular_shield/module/node) in connected_turf)
+
+	if(connected_node)
+
+		connected_node.connected_through_us |= (src)
+		shield_generator = connected_node.shield_generator
+		if(shield_generator)
+			shield_generator.connected_modules |= (src)
+			balloon_alert(user, "connected to generator through node")
+			update_icon_state()
+			if(istype(src, /obj/machinery/modular_shield/module/node))
+				var/obj/machinery/modular_shield/module/node/connected_node = src
+				connected_node.connect_connected_through_us()
+			shield_generator.calculate_boost()
+			return
+		balloon_alert(user, "connected to node")
+		return
+	balloon_alert(user, "failed to find connection")
+
+
+
+/obj/machinery/modular_shield/module/node
+
+	name = "Modular Shield Node"
+	desc = "A waist high mess of humming pipes and wires that extend the modular shield network."
+	icon = 'icons/obj/machines/modular_shield_generator.dmi'
+	icon_state = "node_off_closed"
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * 0.5
+	circuit = /obj/item/circuitboard/machine/modular_shield_node
+	///The list of machines that are connected to us and want connection to a generator
+	var/list/connected_through_us = list()
+
+/obj/machinery/modular_shield/module/node/update_icon_state()
+	. = ..()
+	if(!shield_generator || (machine_stat & NOPOWER))
+		icon_state = "node_off_[panel_open ?"open":"closed"]"
+		return
+	icon_state = "node_on_[panel_open ?"open":"closed"]"
+
+/obj/machinery/modular_shield/module/node/setDir(new_dir)
+	. = ..()
+
+	disconnect_connected_through_us()
+	if(shield_generator)
+		shield_generator.connected_modules -= (src)
+		shield_generator.calculate_boost()
+		shield_generator = null
+		update_icon_state()
```

Comment:
```suggestion
	if(isnull(shield_generator))
		return

	shield_generator.connected_modules -= (src)
	shield_generator.calculate_boost()
	shield_generator = null
	update_icon_state()
```

doing this makes the intent of this proc much more explicit while leveraging the speed and powah of `isnull()`